glean-core/python/glean/glean.py

```diff
@@ -132,56 +132,60 @@ def initialize(
 
         cls._upload_enabled = upload_enabled
 
-        cfg = _ffi.make_config(
-            cls._data_dir,
-            application_id,
-            cls._upload_enabled,
-            configuration.max_events,
-        )
+        # Use `Glean._execute_task` rather than `Glean.launch` here, since we
+        # never want to put this work on the `Dispatcher._preinit_queue`.
+        @Dispatcher._execute_task
+        def initialize():
+            cfg = _ffi.make_config(
+                cls._data_dir,
+                application_id,
+                cls._upload_enabled,
+                configuration.max_events,
+            )
+
+            cls._initialized = _ffi.lib.glean_initialize(cfg) != 0
+
+            # If initialization of Glean fails, we bail out and don't initialize
+            # further
+            if not cls._initialized:
+                return
+
+            for ping in cls._ping_type_queue:
+                cls.register_ping_type(ping)
+
+            # If this is the first time ever the Glean SDK runs, make sure to set
+            # some initial core metrics in case we need to generate early pings.
+            # The next times we start, we would have them around already.
+            is_first_run = _ffi.lib.glean_is_first_run() != 0
+            if is_first_run:
+                cls._initialize_core_metrics()
+
+            # Glean Android sets up the metrics ping scheduler here, but we don't
+            # have one.
+
+            # Deal with any pending events so we can start recording new ones
+            @Dispatcher.launch_at_front
+            def submit_pending_events():
```

Comment:
This is no longer required and may break the init sequence, since it would make ping submission happen _after_ what we're doing right now.

Let's make this consistent with other platforms and [call it directly](https://searchfox.org/glean/rev/6f31bba9617f83f32ba347c6796d816fa3a64cb4/glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt#182-189)

Reply:
Makes sense.